integration-tests/ahm/src/tests.rs

```diff
@@ -344,3 +349,115 @@ async fn migration_works() {
 
 	println!("Migration done in {} RC blocks", rc_block_count);
 }
+
+#[tokio::test(flavor = "current_thread")]
+async fn scheduled_migration_works() {
+	let Some((mut rc, mut ah)) = load_externalities().await else { return };
+
+	// Check that the migration is pending on the RC.
+	rc.execute_with(|| {
+		log::info!("Asserting the initial state on RC");
+		next_block_rc();
+
+		assert_eq!(RcMigrationStageStorage::<Polkadot>::get(), RcMigrationStage::Pending);
+
+		// clear the DMP queue.
+		let _ = DownwardMessageQueues::<Polkadot>::take(AH_PARA_ID);
+	});
+	rc.commit_all().unwrap();
+
+	// Check that the migration is pending on the AH.
+	ah.execute_with(|| {
+		log::info!("Asserting the initial state on AH");
+		next_block_ah();
+
+		assert_eq!(AhMigrationStageStorage::<AssetHub>::get(), AhMigrationStage::Pending);
+
+		// clear the UMP queue.
+		let _ = PendingUpwardMessages::<AssetHub>::take();
+	});
+	ah.commit_all().unwrap();
+
+	// Schedule the migration on RC.
+	let dmp_messages = rc.execute_with(|| {
+		log::info!("Scheduling the migration on RC");
+		next_block_rc();
+
+		let now = frame_system::Pallet::<Polkadot>::block_number();
+		let scheduled_at = now + 2;
+
+		// Fellowship Origin
+		let origin = pallet_xcm::Origin::Xcm(Location::new(
+			0,
+			[
+				Junction::Parachain(1001),
+				Junction::Plurality { id: BodyId::Technical, part: BodyPart::Voice },
+			],
+		));
+		assert_ok!(RcMigrator::schedule_migration(origin.into(), DispatchTime::At(scheduled_at)));
+		assert_eq!(
+			RcMigrationStageStorage::<Polkadot>::get(),
+			RcMigrationStage::Scheduled { block_number: scheduled_at }
+		);
+
+		next_block_rc();
+		// migrating not yet started
+		assert_eq!(
+			RcMigrationStageStorage::<Polkadot>::get(),
+			RcMigrationStage::Scheduled { block_number: scheduled_at }
+		);
+		assert_eq!(DownwardMessageQueues::<Polkadot>::take(AH_PARA_ID).len(), 0);
+
+		next_block_rc();
+
+		// migration started
+		assert_eq!(RcMigrationStageStorage::<Polkadot>::get(), RcMigrationStage::Initializing);
+		let dmp_messages = DownwardMessageQueues::<Polkadot>::take(AH_PARA_ID);
+		assert!(dmp_messages.len() > 0);
+
+		dmp_messages
+	});
+
+	// enqueue DMP messages from RC to AH.
+	ah.execute_with(|| {
+		enqueue_dmp(dmp_messages);
+	});
+	ah.commit_all().unwrap();
+
+	// Asset Hub receives the message from the Relay Chain to start the migration and the
+	// acknowledges it by sending the message back to the Relay Chain.
+	let ump_messages = ah.execute_with(|| {
+		log::info!("Acknowledging the start of the migration on AH");
+		assert_eq!(AhMigrationStageStorage::<AssetHub>::get(), AhMigrationStage::Pending);
+
+		next_block_ah();
+
+		assert_eq!(
+			AhMigrationStageStorage::<AssetHub>::get(),
+			AhMigrationStage::DataMigrationOngoing
+		);
+
+		PendingUpwardMessages::<AssetHub>::take()
+	});
+	ah.commit_all().unwrap();
+
+	// enqueue UMP messages from AH to RC.
+	rc.execute_with(|| {
+		enqueue_ump(ump_messages);
+	});
+	rc.commit_all().unwrap();
```

Comment:
We could make a wrapper for our externalities that always commits after `execute_with`.